### PR TITLE
Start adapting docs for Gradle Kotlin DSL

### PR DESCRIPTION
**Pull request by bartekpacia**
Related to https://github.com/flutter/flutter/issues/151166

Summary of changes made:
- if possible, adapt Groovy syntax to be valid Kotlin DSL syntax
- if above is not possible, add a new tab showing valid Kotlin DSL syntax ([example](https://flutter-docs-prod--pr10846-chore-adapt-for-gradle-kts-cl6muvw5.web.app/add-to-app/android/project-setup))
- also adjusted indentation to be consistently 4 spaces (right now, it's a mix of 2 and 4 in our docs)

I did not go over all Gradle-related content. This is _a start_ to this effort :)

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by sfshaza2 — APPROVED** [review]
This looks ok to me. lgtm

File: src/content/add-to-app/android/plugin-setup.md [hunk 1]
```diff
@@ -127,9 +127,9 @@ in your existing app as follows:
 ```groovy title="ExistingApp/app/build.gradle"
 …
 dependencies {
-  …
-  implementation 'com.google.firebase:firebase-crashlytics:17.0.0-beta03
-  …
+    …
+    implementation("com.google.firebase:firebase-crashlytics:17.0.0-beta0")
```
Comment (domesticmouse) [inline src/content/add-to-app/android/plugin-setup.md:131]: Is `beta03` to `beta0` intentional?

**Review by domesticmouse — APPROVED** [review]

File: src/content/add-to-app/android/plugin-setup.md (hunk 1 above)
Comment (bartekpacia) [reply src/content/add-to-app/android/plugin-setup.md:131]: No, it's a typo! Thanks for the catch

Comment (bartekpacia) [reply src/content/add-to-app/android/plugin-setup.md]: Resolved

**Review by reidbaker — APPROVED** [review]

**Pull request merged**